File: frontend/src/graphql/types/shopping_cart.ts
/** Shopping Cart Details */

import { type ShoppingCart as GraphqlShoppingCart } from '@/graphql/schema/graphql'
import type { LineItems } from '@/graphql/types/line_item'

export type ShoppingCart = {
	lineItems: LineItems
} & Omit<GraphqlShoppingCart, 'lineItems'>
export type ShoppingCarts = ShoppingCart[]

File: backend/app/models/shopping_cart.rb
# typed: strict
# frozen_string_literal: true

# Shopping Cart Details
class ShoppingCart < ApplicationRecord
	has_many :line_items, as: :owner, class_name: 'LineItem', dependent: :delete_all
	belongs_to :session, class_name: 'Session', optional: false, dependent: :delete

	validates_associated :line_items
end
